### PR TITLE
[WIP] Eliminates extra OSD rendering cycles.

### PR DESCRIPTION
**Pull request by mejackreed**
Rather than continuously update redux state that is also passed in as a
prop, this change only makes those redux updates on demand when
explictly called. This increases performance, reduces complexity and
hopefully still accomplishes our state goals. A plugin / or other
component should also be able to explicatly call this instance method if
needed.

Related to #2283 , #2410, #2399 , #2280

### CODE REVIEW

File: src/components/OpenSeadragonViewer.js [hunk 1]
```diff
@@ -73,7 +72,6 @@ export class OpenSeadragonViewer extends Component {
 
     this.osdCanvasOverlay = new OpenSeadragonCanvasOverlay(this.viewer);
     this.viewer.addHandler('update-viewport', this.onUpdateViewport);
-    this.viewer.addHandler('viewport-change', this.onViewportChange);
```
Comment (cbeer) [inline src/components/OpenSeadragonViewer.js]: I don't think `update-viewport` covers all the cases where we need to collect data. At least, when I toggled back and forth between two different windows, I could pretty reliably get the viewer to jump around.

Comment (mejackreed) [reply src/components/OpenSeadragonViewer.js]: I've pushed up an additional commit here that I think addresses the problem. Let me know what you think.

File: src/components/OpenSeadragonViewer.js [hunk 2]
```diff
@@ -125,23 +123,13 @@ export class OpenSeadragonViewer extends Component {
           this.zoomToWorld();
         }
       });
-    } else if (viewer) {
```
Comment (cbeer) [inline src/components/OpenSeadragonViewer.js]: Without this, the zoom controls will break.

**Comment by mejackreed** [comment]
Closing in favor of #2430

**Pull request closed without merging**